### PR TITLE
[settings] subsettings from fake settings 

### PR DESCRIPTION
**Pull request by franramirez688**
Changelog: Bugfix: Avoid `settings.update_values()` failing when deducing compatibility.
Docs: omit

cc @AbrilRBS

### CODE REVIEW

File: test/unittests/model/settings_test.py [hunk 1]
```diff
@@ -481,3 +481,25 @@ def test_settings_intel_cppstd_03():
     settings.compiler = "intel-cc"
     # This doesn't crash, it used to crash due to "03" not quoted in setting.yml
     settings.compiler.cppstd = "03"
+
+
+def test_set_value_non_existing_values():
```
Comment (memsharded) [inline test/unittests/model/settings_test.py:486]: I think I'd add also an integration test. The validation of settings also happens in different places, so better have the full coverage, with ``conan install . -s nonexisting.foo`` test.

File: conans/model/settings.py [hunk 2]
```diff
@@ -302,23 +302,24 @@ def values_list(self):
     def items(self):
         return self.values_list
 
-    def update_values(self, vals):
-        """ receives a list of tuples (compiler.version, value)
-        This is more an updated than a setter
+    def update_values(self, values, raise_error=True):
```
Comment (memsharded) [inline conans/model/settings.py]: I think I'd call it something like ``validate=True``, a bit more meaningful

**Review by memsharded — APPROVED** [review]
Looking good, good job

File: conans/model/settings.py (hunk 2 above)
Comment (franramirez688) [reply conans/model/settings.py]: what about `raise_undefined=True`?

**Review by AbrilRBS — APPROVED** [review]
Nice fix :)

**Pull request merged**